tools/board-runner/src/opentitan.rs

```diff
@@ -0,0 +1,435 @@
+        "examples/tests/console_timeout/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    // Send message
+    p.send_line("Test message")?;
+
+    // Wait 25 seconds
+    let timeout = time::Duration::from_secs(25);
+    thread::sleep(timeout);
+
+    // Send enter
+    p.send_line("")?;
+
+    // Check the message
+    p.exp_string("Userspace call to read console returned: Test message")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn opentitan_malloc_test1() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/malloc_test01/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("malloc01: success")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn opentitan_stack_size_test1() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/stack_size_test01/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("Stack Test App")?;
+    p.exp_string("Current stack pointer: 0x100")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn opentitan_stack_size_test2() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/stack_size_test02/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("Stack Test App")?;
+    p.exp_string("Current stack pointer: 0x100")?;
+
+    Ok(())
+}
+
+fn opentitan_mpu_stack_growth() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_stack_growth/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("This test should recursively add stack frames until exceeding")?;
+    p.exp_string("panicked at 'Process mpu_stack_growth had a fault'")?;
+    p.exp_string("Store/AMO access fault")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn opentitan_mpu_walk_region() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_walk_region/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("MPU Walk Regions")?;
+    p.exp_string("Walking flash")?;
+    p.exp_string("Will overrun")?;
+    p.exp_string("panicked at 'Process mpu_walk_region had a fault'")?;
+
+    Ok(())
+}
+
+fn opentitan_multi_alarm_test() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/multi_alarm_test/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
```

Comment:
it might be good if this script started with a `make clean` of `libtock-c`, and then just matches any `rv32imc.*.tbf` file in each app directory, just so that this script doesn't break next time we need to update the app addresses for opentitan

Reply:
I can't just swap this to `rv32imc.*.tbf` as it then fails to find the elfs.

I'm also not sure that it's a good idea, it's then possible that we will run old elfs. Having to change this when addresses in libtock-c seems somewhat reasonable at least for now based on how much manual intervention is required to run this currently.